### PR TITLE
Kibble cli init

### PR DESCRIPTION
**Pull request by skekre98**
I have currently initialized the cli with `version` and `setup`. I have currently left the setup function open since I believe the set up workflow is a work in progress. Let me know how you'd like me to handle this and what else to build out.

### CODE REVIEW

**Comment by turbaszek** [comment]
> I have currently left the setup function open since I believe the set up workflow is a work in progress.

@skekre98 it would be awesome to rewrite it to using click 👍

**Comment by turbaszek** [comment]
Also the click library should be added to required dependencies in `setup.py`:
```
➜ kibble version
Traceback (most recent call last):
  File "/Users/turbaszek/.pyenv/versions/kibble/bin/kibble", line 33, in <module>
    sys.exit(load_entry_point('apache-kibble', 'console_scripts', 'kibble')())
  File "/Users/turbaszek/.pyenv/versions/kibble/bin/kibble", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/Users/turbaszek/.pyenv/versions/3.8.0/lib/python3.8/importlib/metadata.py", line 75, in load
    module = import_module(match.group('module'))
  File "/Users/turbaszek/.pyenv/versions/3.8.0/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/turbaszek/code/kibble/kibble/__main__.py", line 17, in <module>
    import click
ModuleNotFoundError: No module named 'click'
```
Adding this `click==7.1.2` to setup.py  `INSTALL_REQUIREMENTS` should solve the issue

File: kibble/__main__.py [hunk 1]
```diff
@@ -14,11 +14,24 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+import click
+from version import kibble_version
```
Comment (turbaszek) [inline kibble/__main__.py:18]: Also please use absolute import `from kibble.version import kibble_version`:
```
➜ kibble version
Traceback (most recent call last):
  File "/Users/turbaszek/.pyenv/versions/kibble/bin/kibble", line 33, in <module>
    sys.exit(load_entry_point('apache-kibble', 'console_scripts', 'kibble')())
  File "/Users/turbaszek/.pyenv/versions/kibble/bin/kibble", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/Users/turbaszek/.pyenv/versions/3.8.0/lib/python3.8/importlib/metadata.py", line 75, in load
    module = import_module(match.group('module'))
  File "/Users/turbaszek/.pyenv/versions/3.8.0/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/turbaszek/code/kibble/kibble/__main__.py", line 18, in <module>
    from version import kibble_version
ModuleNotFoundError: No module named 'version'
```

File: kibble/__main__.py [hunk 2]
```diff
@@ -14,11 +14,24 @@
...
 
 
-def main():
-    print("Hello to kibble!")
```
Comment (turbaszek) [inline kibble/__main__.py:20]: By removing `main` you had to adjust the setup.py which uses this function as entry point:
```
➜ kibble version
Traceback (most recent call last):
  File "/Users/turbaszek/.pyenv/versions/kibble/bin/kibble", line 33, in <module>
    sys.exit(load_entry_point('apache-kibble', 'console_scripts', 'kibble')())
  File "/Users/turbaszek/.pyenv/versions/kibble/bin/kibble", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/Users/turbaszek/.pyenv/versions/3.8.0/lib/python3.8/importlib/metadata.py", line 77, in load
    return functools.reduce(getattr, attrs, module)
AttributeError: module 'kibble.__main__' has no attribute 'main'
```

I would suggest to leave this function as in future we may need to do more than just starting the cli. For now I would suggest:
```py
def main():
    cli()
```

**Review by turbaszek — CHANGES_REQUESTED** [review]
The current changes don't work. Please see the comments. Once fixed it should look good!
```
➜ kibble
Usage: kibble [OPTIONS] COMMAND [ARGS]...

  A simple command line tool for kibble

Options:
  --help  Show this message and exit.

Commands:
  setup    starts the setup process for kibble
  version  displays the current kibble versio
```

**Comment by skekre98** [comment]
Hi @turbaszek, how are you building the actual cli? I was testing by running `python __main__.py` 😅. In other words how can I run `kibble version` or other commands with `kibble`?

**Comment by turbaszek** [comment]
> Hi @turbaszek, how are you building the actual cli? I was testing by running `python __main__.py` 😅. In other words how can I run `kibble version` or other commands with `kibble`?

Here are basic information:
https://github.com/apache/kibble/blob/main/CONTRIBUTING.md#development-installation

You need to install `kibble` package, the `-e` flag is for editable installation.

**Comment by skekre98** [comment]
@turbaszek apologies for the delay. I've been able to successfully migrate `save_config` from `setup.py` to `__main__.py` by using KibbleConfigParser. Will work on implementing `create_index` using `click` as well. Let me know if there is anything else you would like to see as well.

**Comment by skekre98** [comment]
This is the current output with the cli:
```bash
(home) Sharvils-MacBook-Pro:kibble sharvilkekre$ kibble
Usage: kibble [OPTIONS] COMMAND [ARGS]...

  A simple command line tool for kibble

Options:
  --help  Show this message and exit.

Commands:
  setup    starts the setup process for kibble
  version  displays the current kibble version
(home) Sharvils-MacBook-Pro:kibble sharvilkekre$ kibble version
1.0.0dev
(home) Sharvils-MacBook-Pro:kibble sharvilkekre$ kibble setup
Welcome to the Apache Kibble setup script!
Elasticsearch: elasticsearch:9200
~/projects/kibble/kibble/api/yaml/kibble.yaml already exists! Writing to ~/projects/kibble/kibble/api/yaml/kibble.yaml.tmp instead
Writing Kibble config to ~/projects/kibble/kibble/api/yaml/kibble.yaml.tmp

All done, Kibble should...work now :)
```

This is currently only missing `create_index`

File: setup.py [hunk 3]
```diff
@@ -38,6 +38,7 @@
     "python-dateutil==2.8.1",
     "PyYAML==5.3.1",
     "tenacity==6.2.0",
+    "click==7.1.2",
```
Comment (turbaszek) [inline setup.py]: Let's keep this list in alphabetical order 👌

**Comment by turbaszek** [comment]
@skekre98 could you please rebase?

File: kibble/__main__.py [hunk 4]
```diff
@@ -14,10 +14,104 @@
...
+    conf.read("kibble.ini")
+    conn_uri = conf.get("elasticsearch", "conn_uri")
+    click.echo(f"Elasticsearch: {conn_uri}")
+
+    # Create Kibble configuration file
+    save_config(
+        mlserver=conf.get("mail", "mailhost"),
+        conn_uri=conn_uri,
+        dbname=conf.get("elasticsearch", "dbname"),
+    )
```
Comment (turbaszek) [inline kibble/__main__.py]: The setup command should do much more than this (especially that after #83 we can abandon saving the config). The `setup` command should perform whole logic of:
https://github.com/apache/kibble/blob/6959f3c51a594941abc08face115cf3057aaed88/kibble/setup/setup.py#L233-L276

Also it should parse the arguments using click not argparse - that's the main point of using click 👍

Comment (skekre98) [reply kibble/__main__.py]: I see, so is it taking in more arguments than what is in `kibble.ini`?

Comment (turbaszek) [reply kibble/__main__.py]: The argparser is implementing a cli. But we would like to migrate to click as it's easier to use. So we need to preserve whole logic of this argument parser (cli):
https://github.com/apache/kibble/blob/6959f3c51a594941abc08face115cf3057aaed88/kibble/setup/setup.py#L41-L87

So we need to use `@click.option` decorators for every option in existing cli

Comment (skekre98) [reply kibble/__main__.py]: Oh got it. So we are not trying to use KibbleConfigParser? Apologies for the misunderstanding.

Comment (turbaszek) [reply kibble/__main__.py]: @skekre98 we are using it to set default values like that:
https://github.com/apache/kibble/blob/6959f3c51a594941abc08face115cf3057aaed88/kibble/setup/setup.py#L47

We don't have to create `KibbleConfigParser` instance every time we want to access it. It's enough to use the `conf` object:
https://github.com/apache/kibble/blob/6959f3c51a594941abc08face115cf3057aaed88/kibble/configuration.py#L32-L33

**Comment by turbaszek** [comment]
Hey @skekre98 any update on this one?

**Comment by skekre98** [comment]
Hi @turbaszek, this is still in progress. I have been caught up with work the past few days.

**Comment by skekre98** [comment]
Got it, I've noticed this PR has quite a lot of fluff so I'm going to go ahead and close this one and open a new one with the correct changes. Apologies for the misunderstanding 😅

**Pull request closed without merging**